client/src/partials/templates/modals/confirmDeletion.modal.js

```diff
@@ -0,0 +1,69 @@
+angular.module('bhima.controllers')
+.controller('ConfirmDeletionModalController', ConfirmDeletionModalController);
+
+// dependencies injection
+ConfirmDeletionModalController.$inject = ['$uibModalInstance', '$translate', 'data'];
+
+/**
+ * Confirm Deletion Controller
+ * This controller is responsible for check a match text given to continue
+ * the deletion process
+ */
+function ConfirmDeletionModalController (Instance, $translate, Data) {
+  var vm = this;
+
+  // Global objects
+  vm.pattern = Data.pattern;
+  vm.elementName = Data.elementName;
+
+  // expose functions to the view
+  vm.close = close;
+  vm.accept = accept;
+
+  // startup
+  startup();
+
+  /** actions at start */
+  function startup() {
+
+    // Confirmation name message
+    var message = $translate.instant('FORM.DIALOGS.PLEASE_TYPE_TEXT');
+
+    vm.message = Data.elementName ?
+      message.replace('%ELEMENT%', Data.elementName) :
+      message.replace('%ELEMENT%', 'element');
+  }
+
+  /** matching */
+  function isMatching(text, pattern) {
+    return pattern && text === pattern;
+  }
+
+  /** close the modal instance*/
+  function close() {
+    Instance.close();
+  }
+
+  /** validation message */
+  function validate(form) {
+    vm.errorMessage = form.text.$invalid && form.$submitted;
+    vm.warningMessage = form.text.$valid && form.$submitted && vm.noCorrespondancy;
```

Comment:
Since these are `Boolean` values, in the future, it makes more sense to name them something like `hasWarningMessage`, as if responding to the question "Does it have a warning message?" (the answer must be yes or no).